### DISCUSSION TITLE
How to set HGHG correcly

### DISCUSSION THREAD

**Discussion by lxdmay** [svenreiche/Genesis-1.3-Version4#172] [Q&A] [answered]
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/138555967/1092be5b-0d2e-452b-91b2-6f3442a0ffb4)
I want to set HGHG schemes(picture above), and  a laser seed induces an energy modulation in the electron beam in the first undulator. when I simulate the schemes, but I can't see the energy modulation in figure.
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/138555967/8478e9ca-a693-4d57-9c28-956b5b8e27ff)
but according to the HGHG theory, the energy modulation is like,
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/138555967/19f248d6-4af0-45b9-9ff2-93a74bebef64)
the input files is in here, I don't know where I set wrongly.
[Desktop.zip](https://github.com/svenreiche/Genesis-1.3-Version4/files/15346151/Desktop.zip)

**Comment by svenreiche** [comment]
Here are a few tips and comments.

First, the second plot is the average energy deviation. If you induce a sinusoidal energy modulation the net change is zero and that is what you can see in the plot. A better indicator would be to plot the energy spread which should grows and - if dominated - by the energy modulation then the rms energy spread should be DE/sqrt(2) where DE is the energy modulation amplitude (e.g.  about 1.7 from the third plot)
Second, if you want to see the actual energy modulation you must dump the particle distribution and then plot the data.  This sinusoidal energy modulation is not given in the main output file of Genesis
third, I would recommend to start with a stead-state simulation to set up the simulation deck. In your input file you have a 50 period modulator but only a time-window of 7 slices. The initial radiation will slip out quickly and then for most of the distance of tracking there is no actual radiaiton fiel dto modulate.  TO have good time-dependent simulattion, your time window should be at lest 4 times the rms duration of the seed puylse + the total slippage length. And the profile for the seeding should be then placed about 2 sigmas away from the end (s0 ~ 2* sig). In steady-state simulation you do not have problem with the slippage so it is much easier (and faster) to set-up the simulation

Hope this helps,
Sven

**Reply by lxdmay** [reply]
thank you,  according to your advise, I simulate a stead-state simulation, the result is here,
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/138555967/4bb756af-1f2a-49b4-ba98-5f4ff95a8577)
but the result is not a sinusoidal energy modulation and the  energy modulation amplitude is very small . is my power not set correctly?

**Reply by lxdmay** [reply]
[Desktop.zip](https://github.com/svenreiche/Genesis-1.3-Version4/files/15367436/Desktop.zip)
this is my input files.

**Reply by svenreiche** [reply, marked as answer]
Again, you are plotting the mean energy. With an energy modulation you would see anything beside some numerical noise, since the energy gain of some electrons are compensated by the energy loss of those 180 degree apart.
Also you do the steady-state simulation but still uses profiles. So your radiaiton and current is evaluated at s=0, which means about 3 sigma away from the peak current and radiation power

**Reply by lxdmay** [reply]
thank you very much. later, I simulate the schemes in a time-dependent simulattion, I have modified the "time" module and the "radiation" module like,
&time
slen=36e-4 
sample=1
&end

&profile_gauss
label = power
c0 = 0.5e6        #W
s0 = 15.278e-4    
sig = 7.639e-4  # m
&end

&field
power = @power  
&end
the energyspread and s is like,
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/138555967/497e86d5-5c29-4adc-ad5e-08a29a83fff3)
the energy and s is like,
![image](https://github.com/svenreiche/Genesis-1.3-Version4/assets/138555967/fbb68569-a3f3-4bd3-a25b-2172b0c66088)
the question is,
(1) if I want to adjust the amplitude, is modifying the c0(is power value) ?
(2) if I want to see the actual energy and s, I should dump the particle distribution ? but the result has a big data.

**Reply by lxdmay** [reply]
> Again, you are plotting the mean energy. With an energy modulation you would see anything beside some numerical noise, since the energy gain of some electrons are compensated by the energy loss of those 180 degree apart. Also you do the steady-state simulation but still uses profiles. So your radiaiton and current is evaluated at s=0, which means about 3 sigma away from the peak current and radiation power

Is there an example of input files for HGHG?

**Reply by svenreiche** [reply]
Hi, if you want to see the growing energy modulation you can plot the energy spread, since this is strongly coupled to the periodic energy modulation, which you do not see, when plotting the mean energy per slice.

And unfortunately, I haven't yet made an example for HGHG simulation. It is on my todo list.

**Reply by lxdmay** [reply]
> Hi, if you want to see the growing energy modulation you can plot the energy spread, since this is strongly coupled to the periodic energy modulation, which you do not see, when plotting the mean energy per slice.
> 
> And unfortunately, I haven't yet made an example for HGHG simulation. It is on my todo list.

Okay, I will wait for your reply.
